pkg/asset/cluster/tfvars.go

```diff
@@ -2,13 +2,13 @@ package cluster
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"os"
 	"strings"
 
-	igntypes "github.com/coreos/ignition/config/v2_2/types"
+	"github.com/clarketm/json"
```

Comment:
And if this is required for ignition and to use special parser, i would recommend we add `Unmarshal` / `Marshal` function to `github.com/openshift/installer/pkg/asset/ignition` so that people can correctly use it if needed.

because I'm pretty sure people will forget.

Reply:
It's more than just cosmetic, although so far we haven't hit the issue in any of the ignition configs created in the code in installer. The problem is that in spec3 some of the optional pointer-struct fields cannot be empty if they are defined, e.g. if `ignition.config.replace` exists,`ignition.config.replace.source` must also exist - otherwise it will fail OpenAPI validation.

See https://github.com/openshift/machine-config-operator/pull/1677

I'll explain this in the commit message.

Reply:
We only really need the `Marshal` function from that package - unmarshaling works fine here with the standard one.